details/impersonation.md

```diff
@@ -0,0 +1,144 @@
+
+```yaml
+readonlyrest:
+  
+  access_control_rules:
+
+    [...]
+
+  impersonation:
+
+    - impersonator: admin1      // Who can impersonate? (user name or pattern)
+      users: ["*"]              // Who can be impersonated? (user names or patterns)
+      auth_key: admin1:pass     // Authentication rule required to impersonate (any authentication rule can be used here)
+
+    - impersonator: admin2
+      users: ["dev2"]
+      ldap_authentication: "ldap1"
+```
+
+In the example above, we see that we have two impersonators: `admin1` and `admin2`. The first one can impersonate any user (`*`) and they are able to authenticate using basic auth (`admin1:pass`). The second impersonator can impersonate only `dev2` user. They will be authenticated using `ldap1` connector.
+
+When an impersonator passes wrong credentials ROR will tell Kibana that impersonation is not allowed.
+
+#### Defining mocks of the external services (optional)
+
+ROR has many sophisticated authentication & authorization methods. Some of them are based on external systems like LDAP. The problem with such systems, in regard to to the impersonation feature, is that those systems either don't support it by default or don't support it at all and even if they do - the configuration is complex.
+
+That's why we decided to solve it totally differently - using mocks. [Wikipedia](https://en.wiktionary.org/wiki/mock) defines `mock` as `an imitation, usually of lesser quality.` And in the case of external authentication systems we are going provide an imitation of it that will tell ACL which users should be successfully authenticated by it. When we consider an authorization service, a mock of it will return the ACL users with their roles in the service. And this is enough for ROR to support impersonation. 
+
+How does ROR use the mocks? Let's suppose we have an `ldap_auth` rule. When ROR processes the rule, it:
+* asks the given LDAP service if the username can be authenticated with a given password, and if they can ...
+* asks LDAP to list what groups the user belongs to
+
+In the impersonation case, it looks pretty much the same. The difference being that ROR won't call any LDAP server - the mock will provide the required information instead (no password required). During impersonating, when ROR processes an LDAP rule, it:
+* asks the mock if the username exists, and if it does ...
+* asks the mock to tell what groups the user belongs to
+
+**⚠️ IMPORTANT:** If one or more of the external services are not mocked, ROR might inform Kibana that the impersonation is not supported. It's better to always define all mocks, to avoid the "Impersonation not supported" Elasticsearch response.
+
+How to define mocks in Kibana is described [here](../examples/impersonation/external-services-mocks-ui.md).
```

Comment:
https://www.w3.org/QA/Tips/noClickHere

Reply:
I'm not sure if I "feel" the rule. But I've changed all the places with the "here link". Please let me know if it's ok now or maybe you have an idea on how to formulate it in a better way

Reply:
The clickable text on a hyperlink should give sufficient context to the reader when they scan a paragraph. When the link text says "click here", the user is forced to rebuild the context from the sentences around the actual link instead, and the whole UX deteriorates.

When you go through documentation especially, you are not reading sequentially. You are scanning for information because you have a problem to solve. So if we are forced to hide information inside a different page, the user at least needs to know quickly what's behind the link before they decide to invest into changing page (and losing the current context).